java/arcs/android/common/resurrection/ResurrectionRequest.kt

```diff
@@ -37,7 +37,12 @@ data class ResurrectionRequest(
      * If empty, the client requests resurrection for *any* change to *any* [StorageKey]-identified
      * data.
      */
-    val notifyOn: List<StorageKey> = emptyList()
+    val notifyOn: List<StorageKey> = emptyList(),
+    /**
+     * Used to associate [notifierId] with a set of [StorageKeys] and is sent back to listeners
+     * on resurrection along with the keys.
+     */
+    val notifierId: String
```

Comment:
I think this makes sense to move above the `notifyOn` value in the order of arguments for ResurrectionRequest's default constructor.